### DISCUSSION TITLE
I created a Face Expression Editor (Import it as a Graph)

### DISCUSSION THREAD

**Discussion by patrickchamp** [Acly/krita-ai-diffusion#1372] [Show and tell]
**What it Does**
Edit facial features, such as eye direction, smile, eyebrow position, rotate head, mouth (Ooh! Eee! Aah!).

<img width="319" alt="image" src="https://github.com/user-attachments/assets/15c56d6e-b713-4957-bc54-f462618c54b8">

**How to use it**
1. Setup: In ComfyUI, install the nodes below. In Krita AI Diffusion, switch from Generate mode to Graph, then import my workflow.
2. Select an individual face to edit: 0 is the 1st face, 1 is the 2nd. (Autodetects up to 50 faces)
3. (Recommended) Select "Generate Live" to see adjustments in real time. It's fast.

Resources: See the AdvancedLivePortrait repo for info about the parameters, I kept the same names. 

**ComfyUI Nodes Used**
Core Functionality: [AdvancedLivePortrait](https://github.com/PowerHouseMan/ComfyUI-AdvancedLivePortrait)
Secondary: [AutoCropFaces](https://github.com/liusida/ComfyUI-AutoCropFaces), [was-node-suite's Image Paste Face Crop](https://github.com/WASasquatch/was-node-suite-comfyui)

**Download Workflow**
[Krita Expession Editor.json](https://github.com/user-attachments/files/17684988/Krita.Expession.Editor.json)

**Comment by artemcrum** [comment]
Fantastic job! Thank you!

Is it possible to add a reset button? When adjusting several face you should to reset slider manually for each face, which is not very convenient.

**Reply by patrickchamp** [reply]
I provided feedback on the Custom Workflow feature and mentioned that a reset to default button would be helpful. There might be a way to do it within the workflow and expose the parameter in krita but you'll need to figure it out yourself. I don't want to set the expectation that i'll update it. i made it for myself and just figured i'd share it. Join the discord though because there's a workflow channel with other people discussing and sharing workflows. Someone else actually created one for the exact same node.

**Reply by Sorry-Im-Stupid** [reply]
You could possibly make it work within your own workflow by having a button for each slider that sets the value to '0'.

**Comment by artemcrum** [comment]
And now FLUX generation is broken(

Here is the error:

> `newbyteorder` was removed from the ndarray class in NumPy 2.0. Use `arr.view(arr.dtype.newbyteorder(order))` instead.

> File "D:\AI\krita\server-new\python\Lib\site-packages\gguf\gguf_reader.py", line 151, in _get                                                                                                                        .newbyteorder(override_order or self.byte_order)

**Reply by artemcrum** [reply]
> And now FLUX generation is broken(

Fixed by updating GGUF custom node in ComfyUI.

**Reply by patrickchamp** [reply]
That's great! Glad i could help. (i had no idea how i was going to help you with that)

**Comment by Sorry-Im-Stupid** [comment]
Very neat, but I've immediately encountered some issues.
The sliders are all out of order on mine, the 'aaa' slider is at the top for some reason and it's not the only one out of place.
And as I was playing with the sliders in live mode I eventually got this:
`Server execution error: {'type': 'prompt_outputs_failed_validation', 'message': 'Prompt outputs failed validation', 'details': 'Value -7.08 smaller than min of 0: wink', 'extra_info': {}} (Error transferring http://127.0.0.1:8188/prompt - server replied: Bad Request)`
And then refused to do anything, had to restart Krita, and when I tried it again I got:
`Server execution error: not enough values to unpack (expected 2, got 1)`
Update: Tried it again and it seems the 'wink' is the only one that causes the initial error, so that's where you should look first.

**Comment by jordimorardo** [comment]
"Setup: In ComfyUI, install the nodes below. In Krita AI Diffusion, switch from Generate mode to Graph" :I don't know  how to do this. I have downloaded the githup repositories of the 3 custom nodes needed and for example, I copied  the liveportrait files in my folder AIdiffusion\ComfyUI\ComfyUI\custom_nodes\ComfyUI-AdvancedLivePortrait, and the same with the other 2, but when I tried of import the workflow I obtenied this error: 
![image](https://github.com/user-attachments/assets/7689510d-99da-44b9-8ced-9b8ec681065e)

Can you help me please ;) ?
